server/src/main/java/server/bazel/WorkspaceTree.java

```diff
@@ -0,0 +1,70 @@
+package server.bazel;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+public class WorkspaceTree<T> {
```

Comment:
This seems like it's also domain, but someone fight me on this

Reply:
I mean, its more of a data structure for the cache rather than a domain object like `Package` or `SourceFile`. That was my thinking.

Reply:
It seems like it's all part of the same data structure to me, because we'll have querying that looks like this ish

```
Workspace.findPackage(...).findTarget(...)
```

In this case, both the `WorkspaceTree` and `Package` are queryable. Maybe it ought to all go in a package called `server.bazel.tree`? That way its all in one spot and still reusable

Reply:
I like that. Moved it there.